homeassistant/components/husqvarna_automower/sensor.py

```diff
@@ -0,0 +1,162 @@
+        key="total_searching_time",
+        translation_key="total_searching_time",
+        icon="mdi:clock-outline",
+        entity_registry_enabled_default=True,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+    ),
+    SensorEntityDescription(
+        key="number_of_charging_cycles",
+        translation_key="number_of_charging_cycles",
+        icon="mdi:battery-sync-outline",
+        entity_registry_enabled_default=True,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="number_of_collisions",
+        translation_key="number_of_collisions",
+        icon="mdi:counter",
+        entity_registry_enabled_default=True,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="total_drive_distance",
+        translation_key="total_drive_distance",
+        entity_registry_enabled_default=True,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.DISTANCE,
+        native_unit_of_measurement=UnitOfLength.METERS,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up sensor platform."""
+    coordinator: AutomowerDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    _LOGGER.debug("coordinator: %s", coordinator.data)
+
+    sensors = [
+        AutomowerSensorEntity(mower_id, coordinator, description)
+        for mower_id in coordinator.data
+        for description in COMMON_SENSOR_TYPES
+    ]
+    statistic_sensors = [
+        AutomowerSensorEntity(mower_id, coordinator, description)
+        for mower_id in coordinator.data
+        for description in SENSOR_TYPES
+        if getattr(coordinator.data[mower_id].statistics, description.key) is not None
+    ]
+    sensors.extend(statistic_sensors)
+    async_add_entities(sensors)
+
+
+class AutomowerSensorEntity(SensorEntity, AutomowerBaseEntity):
+    """Defining the Automower Sensors with SensorEntityDescription."""
+
+    def __init__(
+        self,
+        mower_id: str,
+        coordinator: AutomowerDataUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Set up AutomowerSensors."""
+        super().__init__(mower_id, coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{self.mower_id}_{description.key}"
+
+    @property
+    def native_value(self) -> str:
+        """Return the state of the sensor."""
+        attributes = self.mower_attributes
+        if self.entity_description.key == "battery_percent":
```

Comment:
Please move the value lookup to an entity description function instead.

Example:
https://github.com/home-assistant/core/blob/77f416fa0458b6bfd054256f661fc5acf05c77ca/homeassistant/components/wled/sensor.py#L39

https://github.com/home-assistant/core/blob/77f416fa0458b6bfd054256f661fc5acf05c77ca/homeassistant/components/wled/sensor.py#L50

https://github.com/home-assistant/core/blob/77f416fa0458b6bfd054256f661fc5acf05c77ca/homeassistant/components/wled/sensor.py#L157-L160